src/Orchard.Web/Modules/Orchard.MediaLibrary/Controllers/FolderController.cs

```diff
@@ -62,17 +63,21 @@ IMediaLibraryService mediaManagerService
             }
 
             try {
-                _mediaLibraryService.CreateFolder(viewModel.FolderPath, viewModel.Name);
-                Services.Notifier.Information(T("Media folder created"));
+                bool valid = String.IsNullOrWhiteSpace(viewModel.Name) || Regex.IsMatch(viewModel.Name, @"^[^:?#\[\]@!$&'()*+,.;=\s\""\<\>\\\|%]+$");
+                if (!valid) {
+                    throw new ArgumentException(T("Folder contains invalid characters").ToString());
```

Comment:
Won't it be double-encoded? Can you try with some text that contains HMTL elements?